API

CasinoRoyal666 opened the pull request:
feat: views and urls were set up

CasinoRoyal666 committed:
feat: Add a views and urls, fix a OrderCreateSerializer in serializers.py, update a README.md

valeryia-rusakova commented on events/views.py:
```diff
@@ -1,3 +1,147 @@
+            event = item_data['event']
+            quantity = item_data['quantity']
+
+            if event.available_tickets < quantity:
```
it's better to check availability of all order items before creating order

CasinoRoyal666 committed:
feat: Add CartViewSet into views to work with cart

CasinoRoyal666 committed:
Static/media roots set in the settings file

CasinoRoyal666 committed:
feat: Add API documentation using drf-spectacular (requirements.txt updated)

CasinoRoyal666 committed:
fix: views.py : Methods destroy and clear are fixed in CartView set. Method create in OrderViewSet now creating order in one time

CasinoRoyal666 committed:
feat: New structure of the project:
- Now the serializers, services and views are located in namesake folders
- Every folder contains files with code for Event, Order and Cart models
- Fix the test_order_serializer in accordance with the new model
- Fix the urls.py in accordance with the new structure
- Business logic of Orders, Events and Cart transformed to services files

CasinoRoyal666 committed:
fix: Fix the urls fo the project:
- event-images and order-items deleted
- post and delete methods for event-images now located in EventViewSet class
fix init.py file in views folder
- added docstrings for all services, serializers and views

CasinoRoyal666 committed:
feat: Add availability check in create_order_with_items method
